code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -61,6 +66,7 @@ private Token createToken(ApiRequest request) {
             throw new ClientVisibleException(ResponseCodes.SERVICE_UNAVAILABLE,
                     "codeTypeInvalid", "Code type provided invalid.", null);
         }
+        token.setJwt(authTokenDao.createToken(token.getJwt(), token.getAuthProvider(), ((Policy) ApiContext.getContext().getPolicy()).getAccountId()).getKey());
```

Comment:
we pass the key because vincent will send me the key and i use the key to get the jwt from db. See io.cattle.platform.iaas.api.auth.TokenUtils#retrieveJwt